lib/middleware/httpRequestAllowed.ts

```diff
@@ -8,8 +8,7 @@ const isRequestAllowed = (
   return async (req: NextApiRequest, res: NextApiResponse): Promise<unknown> => {
     try {
       const session = await getSession({ req });
-      const requestBody = req.body && Object.keys(req.body).length > 0 ? req.body : undefined;
-      const method = requestBody?.method ? requestBody.method : req.method;
+      const method = req.method ?? "";
```

Comment:
A little outside the scope of this PR, but your change highlights this.

If `method` ends up being "", then it will check for a session and allow the request to proceed, as long as they are logged in.

If `req.method` ends up being "", I think the default behaviour should either be:
- return a 403, or
- assume a `GET` request, not requiring a `session`

Reply:
I don't think it's outside the scope of this PR at all.  I like the idea of adding more error handling to the `req.method` check.  I'd recommend throwing an error which then triggers the 500 response.   If `req.method` is undefined it's not the clients issues but instead the middleware was not implemented properly.
I'll modify and update this branch.